### ISSUE TITLE
Patches failing unexpectedly/unnecessarily on upgrades.

### ISSUE THREAD

**Issue by dawnmist** [ds300/patch-package#55]
I'm currently using patch-package 6.0.0-5 on Ubuntu Linux 16.04 with node v8.11.1 and yarn 1.6.0.

I am finding that when a package is upgraded using patch-package 6.0.0-5 the patch for the previous version almost always fails to apply, and reports that there was a conflict. In the v5.x.x series of patch-package most patches were still able to be applied to the next version.

Usually, the failed patch applies with no problems using the command-line `patch` command.

I'm attaching a patch for [react-scripts-ts](https://github.com/ds300/patch-package/files/2021637/react-scripts-ts.2.15.1.patch.txt) as an example of one of the unexpected failures. It was made for react-scripts-ts v2.15.1. On upgrading to react-scripts-ts v2.16.0, patch-package reported the following error and refused to use the old patch:
```
**ERROR** Failed to apply patch for package react-scripts-ts

  This error was caused because react-scripts-ts has changed since you
  made the patch file for it. This introduced conflicts with your patch,
  just like a merge conflict in Git when separate incompatible changes are
  made to the same piece of code.

  Maybe this means your patch file is no longer necessary, in which case
  hooray! Just delete it!

  Otherwise, you need to manually fix the patch file. Or generate a new one

  To generate a new one, just repeat the steps you made to generate the first
  one, but accounting for the changes in react-scripts-ts.

  i.e. make changes, run `patch-package react-scripts-ts`, and commit.

  To manually fix a patch file, Run:

     patch -p1 -i patches/react-scripts-ts+2.15.1.patch --verbose --dry-run

  To list rejected hunks. A 'hunk' is a section of patch file that describes
  one contiguous area of changes. They are numbered from 1 and begin with lines
  that look like this:

    @@ -48,5 +49,6 @@ function foo(bar) {

  Remove the conflicting hunks, then manually edit files in

    node_modules/react-scripts-ts

  to reflect the changes that the conflicting hunks were supposed to make.

  Then run `patch-package react-scripts-ts`

  Info:
    Patch was made for version 2.15.1
    Meanwhile node_modules/react-scripts-ts is version 2.16.0

error Command failed with exit code 1.
```
Using `patch` applied the patch with minor line offsets to the second & third chunk:
```
$> patch -p1 -i patches/react-scripts-ts+2.15.1.patch
patching file node_modules/react-scripts-ts/config/paths.js
Hunk #2 succeeded at 72 (offset 1 line).
Hunk #3 succeeded at 105 (offset 2 lines).
```
I know that there was a change in how patches are applied in the v6 branch. I think the above indicates that there may be an issue with the current implementation, as the example patch really should have been able to be applied to the newer react-scripts-ts package version.

**Comment by ds300** [comment]
Hi! :wave: Thanks for the report! This is a known issue and something that I'll fix before releasing 6.0.0.

**Comment by ds300** [comment]
I've taken care of this and just published 6.0.0 :tada:
